fusion/fusion/fusion.h

```diff
@@ -0,0 +1,262 @@
+        //! \brief ...
+        using FusionGroupCfgType =
+            fusion::FusionGroupCfg<MachineInfoType, FieldExtractorType>;
+        //! \brief ...
+        using FusionGroupCfgListType = std::vector<FusionGroupCfgType>;
+        //! \brief ...
+        using TransformFuncType = bool (*)(FusionGroupType &,
+                                           InstPtrListType &,
+                                           InstPtrListType &);
+        //! \brief ...
+        using FusionContextType = fusion::FusionContext<FusionGroupType>;
+
+        //! \brief ...
+        using FusionFuncType = std::function<void(
+            Fusion &, InstPtrListType &, InstPtrListType &)>;
+
+        //! \brief main ctor
+        Fusion(const FusionGroupListType & fusiongroup_list,
+               const FusionGroupCfgListType & fusiongroupcfg_list,
+               const FileNameListType & txt_file_list,
+               const FusionGroupTypeAlloc fusiongroup_alloc =
+                   fusion::ShrPtrAlloc<FusionGroupType>(),
+               const MachineInfoTypeAlloc machine_info_alloc =
+                   fusion::ShrPtrAlloc<MachineInfoType>(),
+               const FieldExtractorType field_extractor_alloc =
+                   fusion::ShrPtrAlloc<FieldExtractorType>()) :
+            fusiongroup_alloc_(fusiongroup_alloc),
+            machine_info_alloc_(machine_info_alloc),
+            fusionOpr(defaultFusionOpr)
+        {
+            if (fusiongroup_list.size() > 0)
+                initialize(fusiongroup_list);
+            else if (fusiongroupcfg_list.size() > 0)
+                initialize(fusiongroupcfg_list);
+            else if (txt_file_list.size() > 0)
+                initialize(txt_file_list);
+            context_.makeContext("fbase", fusiongroup_list);
+        }
+
+        //! \brief ctor from group list
+        Fusion(const FusionGroupListType & fusiongroup_list) :
+            Fusion(fusiongroup_list, {}, {}, FusionGroupTypeAlloc(),
+                   MachineInfoTypeAlloc(), FieldExtractorType())
+        {
+        }
+
+        //! \brief ctor from cfg group list
+        Fusion(const FusionGroupCfgListType & fusiongroupcfg_list) :
+            Fusion({}, fusiongroupcfg_list, {}, FusionGroupTypeAlloc(),
+                   MachineInfoTypeAlloc(), FieldExtractorType())
+        {
+        }
+
+        //! \brief ctor from txt file list
+        //!
+        //! This is for future support of the DSL or JSON.
+        //! The plan a constructor helper determines file type
+        Fusion(const FileNameListType & txt_file_list) :
+            Fusion({}, {}, txt_file_list, FusionGroupTypeAlloc(),
+                   MachineInfoTypeAlloc(), FieldExtractorType())
+        {
+        }
+
+        //! \brief bare ctor
+        Fusion() :
+            Fusion({}, {}, {}, FusionGroupTypeAlloc(),
+                   MachineInfoTypeAlloc(), FieldExtractorType())
+        {
+        }
```

Comment:
Curious, does this ctor have any value?

Reply:
A bare FusionType was instantiated in the testbench compile checks test. Removing the instance eliminates the need for it. 

//Removed FusionType f1;